File: app/src/main/java/com/quigglesproductions/secureimageviewer/glide/CustomGlideModule.java
package com.quigglesproductions.secureimageviewer.glide;

import android.content.Context;

import androidx.annotation.NonNull;

import com.bumptech.glide.Glide;
import com.bumptech.glide.GlideBuilder;
import com.bumptech.glide.Registry;
import com.bumptech.glide.annotation.Excludes;
import com.bumptech.glide.annotation.GlideModule;
import com.bumptech.glide.integration.okhttp3.OkHttpLibraryGlideModule;
import com.bumptech.glide.integration.okhttp3.OkHttpUrlLoader;
import com.bumptech.glide.load.DecodeFormat;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.load.model.GlideUrl;
import com.bumptech.glide.module.AppGlideModule;
import com.bumptech.glide.request.RequestOptions;
import com.quigglesproductions.secureimageviewer.dagger.hilt.annotations.RequestServiceClient;

import java.io.InputStream;

import dagger.hilt.EntryPoint;
import dagger.hilt.EntryPoints;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;
import okhttp3.Call;
import okhttp3.OkHttpClient;

@GlideModule
//@Excludes(OkHttpLibraryGlideModule.class)
public class CustomGlideModule extends AppGlideModule {
    @Override
    public void applyOptions(@NonNull Context context, @NonNull GlideBuilder builder) {
        builder.setDefaultRequestOptions(new RequestOptions()
                .format(DecodeFormat.PREFER_RGB_565)
                .diskCacheStrategy(DiskCacheStrategy.AUTOMATIC)
        );
    }

    @Override
    public void registerComponents(@NonNull Context context, @NonNull Glide glide, @NonNull Registry registry) {
        super.registerComponents(context, glide, registry);
        GlideModuleEntryPoint entryPoint = EntryPoints.get(context, GlideModuleEntryPoint.class);
        OkHttpClient client = entryPoint.provideOkHttpClient();
        registry.replace(GlideUrl.class, InputStream.class, new OkHttpUrlLoader.Factory((Call.Factory) client));
    }

    @EntryPoint
    @InstallIn(SingletonComponent.class)
    interface GlideModuleEntryPoint {

        @RequestServiceClient
        OkHttpClient provideOkHttpClient();
    }
}

File: app/src/main/java/com/quigglesproductions/secureimageviewer/glide/CustomGlideModule.kt
package com.quigglesproductions.secureimageviewer.glide

import android.content.Context
import com.bumptech.glide.Glide
import com.bumptech.glide.GlideBuilder
import com.bumptech.glide.Registry
import com.bumptech.glide.annotation.GlideModule
import com.bumptech.glide.integration.okhttp3.OkHttpUrlLoader
import com.bumptech.glide.load.DecodeFormat
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.load.model.GlideUrl
import com.bumptech.glide.module.AppGlideModule
import com.bumptech.glide.request.RequestOptions
import com.quigglesproductions.secureimageviewer.dagger.hilt.annotations.RequestServiceClient
import dagger.hilt.EntryPoint
import dagger.hilt.EntryPoints
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import okhttp3.OkHttpClient
import java.io.InputStream

@GlideModule
class CustomGlideModule : AppGlideModule() {
    override fun applyOptions(context: Context, builder: GlideBuilder) {
        builder.setDefaultRequestOptions(
            RequestOptions()
                .format(DecodeFormat.PREFER_RGB_565)
                .diskCacheStrategy(DiskCacheStrategy.NONE)
        )
    }

    override fun registerComponents(context: Context, glide: Glide, registry: Registry) {
        super.registerComponents(context, glide, registry)
        val entryPoint = EntryPoints.get(
            context,
            GlideModuleEntryPoint::class.java
        )
        val client = entryPoint.provideOkHttpClient()
        val factory = OkHttpUrlLoader.Factory(client)
        registry.replace(GlideUrl::class.java,InputStream::class.java, factory)
    }

    @EntryPoint
    @InstallIn(SingletonComponent::class)
    internal interface GlideModuleEntryPoint {
        @RequestServiceClient
        fun provideOkHttpClient(): OkHttpClient
    }
}
